service/kernel/oneapi/cl_kernels/radix_sort.cl

```diff
@@ -0,0 +1,144 @@
+/* file: radix_sort.cl */
+/*******************************************************************************
+* Copyright 2014-2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of radix sort kernels.
+//--
+*/
+
+#ifndef __RADIX_SORT_CL__
+#define __RADIX_SORT_CL__
+
+#define DECLARE_SOURCE(name, src) static const char* name = #src;
+
+DECLARE_SOURCE(radix_sort_simd,
+
+void swap(  __global sortedType **input,
+            __global sortedType **output) {
+    __global sortedType *tmp = *input;
+    *input = *output;
+    *output = tmp;
+}
+
+__kernel void radix_sort_group( __global  sortedType      *labels,
+                                __global  int             *sorted,
+                                __global  int             *radixbuf,
+                                int N,
+                                int BlockOffset) {
+
+    const int global_id  = get_global_id(0);
+    const int local_id = get_local_id(1);
+// Added for debug purposes!!! Should be controlled by workgroup size to avoid impact on efficiency!!!!    
```

Comment:
Probably you forget to delete this debug code. We do not want to include debug code into production